internal/statemachine/classic_test.go

```diff
@@ -3191,6 +3215,96 @@ func TestFailedUpdateBootloader(t *testing.T) {
 		saveCWD := helper.SaveCWD()
 		defer saveCWD()
 
+		var stateMachine ClassicStateMachine
+		stateMachine.commonFlags, stateMachine.stateMachineFlags = helper.InitCommonOpts()
+		stateMachine.parent = &stateMachine
+		stateMachine.Snaps = []string{"lxd"}
+		stateMachine.commonFlags.Channel = "stable"
+		stateMachine.ImageDef = imagedefinition.ImageDefinition{
+			Architecture: getHostArch(),
+			Series:       getHostSuite(),
+			Rootfs: &imagedefinition.Rootfs{
+				Archive: "ubuntu",
+			},
+			Customization: &imagedefinition.Customization{
+				ExtraPackages: []*imagedefinition.Package{
+					{
+						PackageName: "squashfs-tools",
+					},
+					{
+						PackageName: "snapd",
+					},
+				},
+				ExtraSnaps: []*imagedefinition.Snap{
+					{
+						SnapName: "hello",
+					},
+					{
+						SnapName: "core",
+					},
+					{
+						SnapName: "core20",
+					},
+				},
+			},
+		}
+
+		err := stateMachine.makeTemporaryDirectories()
+		asserter.AssertErrNil(err, true)
+
+		// create chroot to preseed
+		err = stateMachine.createChroot()
+		asserter.AssertErrNil(err, true)
+
+		// install the packages that snap-preseed needs
+		err = stateMachine.installPackages()
+		asserter.AssertErrNil(err, true)
+
+		// first call prepareClassicImage to eventually preseed it
+		err = stateMachine.prepareClassicImage()
+		asserter.AssertErrNil(err, true)
+
+		// now preseed the chroot
+		err = stateMachine.preseedClassicImage()
+		asserter.AssertErrNil(err, true)
+
+		// set up a new set of snaps to be installed
+		stateMachine.ImageDef = imagedefinition.ImageDefinition{
+			Architecture: getHostArch(),
+			Customization: &imagedefinition.Customization{
+				ExtraSnaps: []*imagedefinition.Snap{
+					{
+						SnapName: "ubuntu-image",
+					},
+				},
+			},
+		}
+
+		// call prepareClassicImage again to trigger the reset
+		err = stateMachine.prepareClassicImage()
+		asserter.AssertErrNil(err, true)
+
+		// make sure the snaps from both prepares are present
+		expectedSnaps := []string{"lxd", "hello", "ubuntu-image"}
+		for _, expectedSnap := range expectedSnaps {
+			snapGlobs, err := filepath.Glob(filepath.Join(stateMachine.tempDirs.chroot,
+				"var", "lib", "snapd", "seed", "snaps", fmt.Sprintf("%s*.snap", expectedSnap)))
+			asserter.AssertErrNil(err, true)
+			if len(snapGlobs) == 0 {
+				t.Errorf("expected snap %s to exist in the chroot but it does not", expectedSnap)
+			}
+		}
+	})
+}
+
+// TestPreseedResetChroot tests that calling prepareClassicImage on a
+// preseeded chroot correctly resets the chroot and preseeds over it
+func TestPreseedResetChroot(t *testing.T) {
```

Comment:
Ah, I think this is the mistake! I think this is supposed to be called `TestFailedUpdateBootloader` and the one above is the `TestPreseedResetChroot` test! Please switch their places ;)